src/Magnum/DartIntegration/Object.cpp

```diff
@@ -62,32 +64,30 @@ Object& Object::update(Trade::AbstractImporter* importer) {
         return *this;
 
     /* Get transform from DART */
-    const Eigen::Isometry3d* trans;
+    Matrix4 trans;
     if(!_node)
-        trans = &_body->getRelativeTransform();
+        trans = Matrix4(Matrix4d(_body->getRelativeTransform()));
     else
-        trans = &_node->getRelativeTransform();
-
-    Eigen::Quaterniond quat(trans->linear());
-    Eigen::Vector3d axis(quat.x(), quat.y(), quat.z());
-    double angle = 2.*std::acos(quat.w());
-    if(std::abs(angle)>1e-5) {
-        axis = axis.array() / std::sqrt(1-quat.w()*quat.w());
-        axis.normalize();
-    }
-    else
-        axis(0) = 1.;
+        trans = Matrix4(Matrix4d(_node->getRelativeTransform()));
 
-    Eigen::Vector3d T = trans->translation();
+    /* Check if any value is NaN */
+    if(Math::isNan(trans[0]).any() || Math::isNan(trans[1]).any() || Math::isNan(trans[2]).any() || Math::isNan(trans[3]).any()) {
+        Warning{} << "DartIntegration::Object::update(): Received NaN values from DART. Ignoring this update.";
+        return *this;
+    }
 
-    /* Convert it to axis-angle representation */
-    Math::Vector3<Float> t(T[0], T[1], T[2]);
-    Math::Vector3<Float> u(axis(0), axis(1), axis(2));
-    Rad theta(angle);
+    Quaternion quat = Quaternion::fromMatrix(trans.rotationScaling());
+    Vector3 axis = quat.axis();
+    Rad angle = quat.angle();
+    if(Math::abs(Float(angle))<=1e-5f) {
+        axis = {1.f, 0.f, 0.f};
+    }
+    axis = axis.normalized();
+    Vector3 t = trans.translation();
 
     /* Pass it to Magnum */
     _transformation.resetTransformation()
-        .rotate(theta, u)
+        .rotate(angle, axis)
```

Comment:
Why go through the pain of creating a quaternion and then again converting that to a transformation matrix inside `rotate()`? Wouldn't just the following work as well?

```cpp
_transformation.setTransformation(trans);
```

Reply:
`error: ‘class Magnum::SceneGraph::AbstractBasicTranslationRotation3D<float>’ has no member named ‘setTransformation’; did you mean ‘resetTransformation’?`

Reply:
blargh, forgot about this :sweat_smile: ... okay then, keep it as-is

Reply:
I guess you should provide this function? What is the rationale of not providing it for AbstractFeature?

Reply:
Because some transformations support only translation, or only translation/rotation, and the full transformation matrix could be too general for these.

... but I should provide `rotate()` taking a quat, definitely, adding that to my TODOs.